src/Upcasting/UpcasterChain.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * This file is part of the prooph/event-store.
+ * (c) 2014-2017 prooph software GmbH <contact@prooph.de>
+ * (c) 2015-2017 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Upcasting;
+
+use Prooph\Common\Messaging\Message;
+
+final class UpcasterChain implements Upcaster
+{
+    /**
+     * @var Upcaster[]
+     */
+    private $upcasters;
+
+    public function __construct(Upcaster ...$upcasters)
+    {
+        $this->upcasters = $upcasters;
+    }
+
+    public function upcast(Message $message): array
```

Comment:
Can you take a second look on the message?

Why is the message turned into a messages array just to foreach over it?

You wanted to return `messages`, right? But `$result` is returned.
Foreach can be removed in either case.